src/control/server/config/server_test.go

```diff
@@ -391,6 +391,18 @@ func TestServerConfig_Validation(t *testing.T) {
 			},
 			expErr: FaultConfigBadControlPort,
 		},
+		"single access point including negative port": {
+			extraConfig: func(c *Server) *Server {
+				return c.WithAccessPoints("1.2.3.4:-10002")
+			},
+			expErr: FaultConfigBadControlPort,
+		},
+		"single access point hostname including negative port": {
+			extraConfig: func(c *Server) *Server {
+				return c.WithAccessPoints("hostX:-10002")
+			},
+			expErr: errors.New("invalid hostname"),
+		},
```

Comment:
Huh, why isn't the fault being returned in this case?

Reply:
This is a feature of the irregularity noted in hostlist, when the host set is created from the access points during validation, the negative port produces a hostname error (rather than succeeding in the IP case, to fail later in the validation).